azure-keyvault-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/boot/AzureKeyVaultCredential.java

```diff
@@ -0,0 +1,43 @@
+package com.microsoft.azure.keyvault.spring.boot;
+
+import com.microsoft.aad.adal4j.AuthenticationContext;
+import com.microsoft.aad.adal4j.AuthenticationResult;
+import com.microsoft.aad.adal4j.ClientCredential;
+import com.microsoft.azure.keyvault.authentication.KeyVaultCredentials;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+
+public class AzureKeyVaultCredential extends KeyVaultCredentials {
+    private String clientId;
+    private String clientKey;
+
+    private static final Integer TOKEN_TIMEOUT = 3;
+
+    public AzureKeyVaultCredential(String clientId, String clientKey) {
+        this.clientId = clientId;
+        this.clientKey = clientKey;
+    }
+
+    @Override
+    public String doAuthenticate(String authorization, String resource, String scope) {
+        AuthenticationContext context = null;
+        AuthenticationResult result = null;
+        String token = "";
+
+        try {
+            context = new AuthenticationContext(authorization, false, Executors.newSingleThreadExecutor());
```

Comment:
no customized authority provided here, but default one. and all examples codes use false. https://azure.github.io/azure-sdk-for-java/com/microsoft/azure/keyvault/authentication/KeyVaultCredentials.html

trusted host actually includes: 
    private final static String[] TRUSTED_HOST_LIST = { "login.windows.net",
            "login.chinacloudapi.cn", "login.cloudgovapi.us" };